LWCOSC/force-app/main/default/lwc/onlineShopIndexPage/onlineShopIndexPage.js

```diff
@@ -0,0 +1,346 @@
+import { LightningElement,track } from 'lwc';
+
+import updatePurchaseOrderRecords from '@salesforce/apex/ProductController.updatePurchaseOrderRecords';
+import Id from '@salesforce/user/Id';
+
+
+export default class OnlineShopIndexPage extends LightningElement {
+    purchaseOrderFlag=true;
+	ispurchaseOrderBtnDisabled = false;
+    productListFlag;
+    isNewPurchaseBtnDisable;
+    cartDataFlag;
+    priorData=[];
+    cartData=[];
+	cartErrors = [];
+	invoiceDataFlag;
+	@track invoiceCartData = [];
+	invoiceDate;
+	invoice;
+	userId = Id;
+
+    historyHandler(event){
+       this.purchaseOrderFlag = event.detail.purchaseOrderFlag;
+       this.productListFlag = event.detail.productListFlag;
+        
+    }
+
+    newPurchaseBtnHandler(){
+		this.ispurchaseOrderBtnDisabled = true;
+        this.productListFlag = true;
+    }
+
+
+    // Start:-  Updation in ProdctListData
+    goToCartClickHandler(event){
+        let selectedProductList = event.detail.selectedProductList;
+        let selectedProductListIds = event.detail.selectedProductListIds;
+        this.priorData = event.detail.priorData;
+        this.cartDataFlag = event.detail.cartDataFlag;
+        this.updateProductData(selectedProductListIds);
+        this.updateCartProductData(selectedProductListIds, selectedProductList);
+		this.template.querySelector('c-product-list-new').setPageData(this.priorData);
+		
+         
+     }
+
+     updateProductData(selectedProductListIds){
+		let newPriorData = [];
+		for(let i=0; i<this.priorData.length;i++){
+			if(selectedProductListIds.includes(this.priorData[i].Id)){
+				let product = new Object();
+				product.Id = this.priorData[i].Id;
+				product.Name = this.priorData[i].Name;
+				product.Available_Quantity__c = Number(this.priorData[i].Available_Quantity__c) -Number(1);
+				product.ProductCode = this.priorData[i].ProductCode;
+				product.Total_Price__c = this.priorData[i].Total_Price__c;
+				newPriorData.push(product);
+			}
+			else{
+				newPriorData.push(this.priorData[i]);
+			}
+		}
+		this.priorData = newPriorData;
+	}
+
+
+	updateCartProductData(selectedProductListIds, selectedProductList){
```

Comment:
this method is similar to the previous method, we could think of a generic one to save lines of code and redundancy

Reply:
Point Noted, As discussion No action require